infrastructure/src/test/java/org/corfudb/infrastructure/log/StreamLogDataStoreTest.java

```diff
@@ -22,23 +22,45 @@
     public void testGetAndSave() {
         StreamLogDataStore streamLogDs = getStreamLogDataStore();
 
-        final int tailSegment = 333;
+        final long headSegment = 555L;
+        streamLogDs.updateHeadSegment(headSegment);
+        assertEquals(headSegment, streamLogDs.getHeadSegment());
+
+        final long tailSegment = 333L;
         streamLogDs.updateTailSegment(tailSegment);
         assertEquals(tailSegment, streamLogDs.getTailSegment());
 
-        final int startingAddress = 555;
-        streamLogDs.updateStartingAddress(startingAddress);
-        assertEquals(startingAddress, streamLogDs.getStartingAddress());
+        final long compactionMark = 666L;
+        streamLogDs.updateGlobalCompactionMark(compactionMark);
+        assertEquals(compactionMark, streamLogDs.getGlobalCompactionMark());
+
+        final long committedTail = 777L;
+        streamLogDs.updateCommittedTail(committedTail);
+        assertEquals(committedTail, streamLogDs.getCommittedTail());
     }
 
     @Test
     public void testReset() {
         StreamLogDataStore streamLogDs = getStreamLogDataStore();
-        streamLogDs.resetStartingAddress();
-        assertEquals(INITIAL_ADDRESS, streamLogDs.getStartingAddress());
+
+        final long initialValue = 100L;
+
+        streamLogDs.updateHeadSegment(initialValue);
+        streamLogDs.updateTailSegment(initialValue);
+        streamLogDs.updateGlobalCompactionMark(initialValue);
+        streamLogDs.updateCommittedTail(initialValue);
+
+        streamLogDs.resetHeadSegment();
+        assertEquals(streamLogDs.getHeadSegment(), Address.MAX);
```

Comment:
It is initialized to MAX so that it can be updated by using min() every time. HeadSegment and tailSegment are inferred from reading the whole address space by repeatedly updating.